experimental/dds/tree2/src/feature-libraries/editable-tree-2/proxies/proxies.ts

```diff
@@ -497,3 +514,94 @@ export function createListProxy<TTypes extends AllowedTypes>(
 		},
 	});
 }
+
+// #region Create dispatch map for maps
+
+const mapStaticDispatchMap: PropertyDescriptorMap = {};
+
+[
+	/* eslint-disable @typescript-eslint/unbound-method */
+	Map.prototype.clear,
+	Map.prototype.delete,
+	Map.prototype.entries,
+	Map.prototype.forEach,
+	Map.prototype.get,
+	Map.prototype.has,
+	Map.prototype.keys,
+	Map.prototype.set,
+	Map.prototype.values,
+	/* eslint-enable @typescript-eslint/unbound-method */
+].forEach((fn) => {
+	listPrototypeProperties[fn.name] = { value: fn };
+});
+
+mapStaticDispatchMap[Symbol.iterator] = {
+	value: Map.prototype[Symbol.iterator],
+	writable: false,
+	enumerable: false,
+	configurable: false,
+};
+
+const mapPrototype = Object.create(Object.prototype, mapStaticDispatchMap);
+
+// #endregion
+
+const getMapNode = <TSchema extends MapSchema>(target: object): MapNode<TSchema> => {
+	return getTreeNode(target) as MapNode<TSchema>;
+};
+
+function createMapProxy<TSchema extends MapSchema>(treeNode: TreeNode): SharedTreeMap<TSchema> {
+	// Create a 'dispatch' object that this Proxy forwards to instead of the proxy target.
+	// Own properties on the dispatch object are surfaced as own properties of the proxy.
+	// (e.g., 'size', which is defined below).
+	//
+	// Properties normally inherited from 'Map.prototype' are surfaced via the prototype chain.
+	const dispatch = Object.create(mapPrototype, {
+		size: {
+			get(this: object) {
+				return getMapNode(this).size;
+			},
+			set() {},
+			enumerable: false,
+			configurable: false,
+		},
+	});
+
+	setTreeNode(dispatch, treeNode);
+
+	// To satisfy 'deepEquals' level scrutiny, the target of the proxy must be an object with the same
+	// 'null prototype' you would get from on object literal '{}' or 'Object.create(null)'.  This is
+	// because 'deepEquals' uses 'Object.getPrototypeOf' as a way to quickly reject objects with different
+	// prototype chains.
+
+	// TODO: Although the target is an object literal, it's still worthwhile to try experimenting with
+	// a dispatch object to see if it improves performance.
+	return new Proxy<SharedTreeMap<TSchema>>(new Map<string, ProxyNode<TSchema>>(), {
+		get: (target, key, receiver): unknown => {
+			console.log(`get "${String(key)}"`);
+
+			if (key === "get") {
+				return (mapKey: string) => {
+					const mapNode = getMapNode(dispatch);
+					return getProxyForField(mapNode.getBoxed(mapKey));
+				};
+			}
```

Comment:
Here you're intercepting "get" and returning your own implementation before looking it up in the dispatch object.

Just move this implementation to the dispatch object's prototype, per the above.

We're essentially using the dispatch object (and prototype) to replace this switch:

switch(key) {
    case "get": ...
    case "set": ...
    case "has": ...
    ...
}

With a "map" of string -> Function.

Reply:
Ah, okay. Makes sense. Will update!